types/block.go

```diff
@@ -229,9 +229,7 @@ func (b *Block) fillDataAvailabilityHeader() {
 	shares := namespacedShares.RawShares()
 
 	if len(shares) == 0 {
-		// no shares -> no row/colum roots -> hash(empty)
-		b.DataHash = b.DataAvailabilityHeader.Hash()
-		return
+		panic("shares cannot be empty")
```

Comment:
Maybe remove the if block entirely?

Reply:
Yeah, that makes sense.

Reply:
The only purpose it serves is documenting that the shares can't be empty. But a) that is not too important to know when going through this part and b) a simple comment would communicate the same thing.

Reply:
https://github.com/lazyledger/lazyledger-core/pull/323/commits/5a910897b1e67b5e2460d1c16431bb28380c21ec